stats/stats.py

```diff
@@ -64,122 +64,112 @@ def __init__(self, queryset, result_type, modifier=None, granulations=None):
-                    if not granulation_parameters:
-                        continue
-                    for granulation in self.granulations:
-                        if granulation['value'] is not None and granulation['value'] not in \
-                                granulation_parameters:
-                            raise ValueError("The granulation parameter is incorrect")
 
-        return self.get_result()
-
-    def get_granulations(self):
-        granulation_arguments = {}
-        for granulation in self.granulations:
-            name = granulation['name']
-            if ((self.queryset.model == Transaction and name in ['created_at', 'updated_at']) or
-                    (self.queryset.model == Invoice and name in ['issue_date', 'paid_date']) or
-                    (self.queryset.model == BillingLog and name == 'plan')):
-                granulation_arguments['granulation_field'] = name
-                if granulation['value'] is not None:
-                    granulation_arguments['time_granulation_interval'] = granulation['value']
-            else:
-                granulation_arguments['additional_granulation_field'] = name
-        return granulation_arguments
-
-    def documents_overdue_payments(self, queryset):
-        stats_list = []
-        for i in queryset.filter(due_date__lte=datetime.now(), state=Invoice.STATES.ISSUED):
-            name = i.customer.first_name + " " + i.customer.last_name
-            new_document = {'customer': name,
-                            'due_date': i.due_date,
-                            'values': []
-                            }
-            data = {
-                'total': i.total,
-                'paid': i.amount_paid_in_transaction_currency,
-                'left': i.amount_to_be_charged_in_transaction_currency
-            }
-            new_document['values'] = data
+                # foreach granulation field verify if it has granulation parameters or if they
+                # are correct
+                for granulation_field, granulation_parameters in values['granulations'].iteritems():
+                    self._validate_granulation_parameters(granulation_field, granulation_parameters,
+                                                          values['granulations'])
 
-            stats_list.append(new_document)
+        return True
 
-        return stats_list
-
-    def _numbers_to_strings(self, argument):
-        switcher = {
-            1: "Jan",
-            2: "Feb",
-            3: "Mar",
-            4: "Apr",
-            5: "May",
-            6: "Jun",
-            7: "Jul",
-            8: "Aug",
-            9: "Sep",
-            10: "Oct",
-            11: "Nov",
-            12: "Dec",
-        }
-        return switcher.get(argument, "none")
+    # iterates the granulated_stats dictionary and creates the stats_list
+    def _create_stats_list(self, granulated_stats, stats_list, granulation_field,
```

Comment:
You don't need to receive `stats_list`, you can just return a new list from this method.